activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -943,6 +944,13 @@ def configure_connection
           self.client_min_messages = @config[:min_messages] || "warning"
           self.schema_search_path = @config[:schema_search_path] || @config[:schema_order]
 
+          @raw_connection.set_notice_receiver do |result|
```

Comment:
Will this be invoked if `ActiveRecord.db_warnings_action.nil?` ?

Reply:
It will. We could avoid setting this if the config is set to ignore -- originally we were (e.g. [here](https://github.com/rails/rails/pull/46690#discussion_r1065906927)) but I think we moved where the receiver was set and dropped the check.

We could open a PR to remove [the early return](https://github.com/rails/rails/pull/46690/files#diff-e6fd03cad5e3437c76b6c5db106359124dc9feab8341ade39b9ae54af001fac9R181) from PG's `#handle_warnings`, and to instead avoid setting the notice receiver here if `ActiveRecord.db_warnings_action.nil?`. What do you think?